pkg/kafsar/kafka_impl.go

```diff
@@ -209,18 +209,18 @@ func (k *KafkaImpl) OffsetListPartition(addr net.Addr, topic string, req *servic
 			ErrorCode: service.UNKNOWN_SERVER_ERROR,
 		}, nil
 	}
-	var message pulsar.Message
-	if req.Time == constant.TimeEarliest {
-		message = utils.ReadEarliestMsg(fullTopicName, k.kafsarConfig.MaxFetchWaitMs, req.PartitionId, k.pulsarClient)
+	offset := constant.DefaultOffset
+	if req.Time == constant.TimeLasted {
+		pulsarUrl := fmt.Sprintf("http://%s:%d", k.pulsarConfig.Host, k.pulsarConfig.HttpPort)
```

Comment:
extract a method of `kafsarImpl` to get pulsarUrl,  easier for others call http api. WDYT?